packages/circuits/noir/crates/smt_pedersen/src/lib.nr

```diff
@@ -0,0 +1,200 @@
+use dep::std::option::Option;
+
+mod utils;
+
+global TREE_DEPTH: u32 = 256; 
+
+/**
+ * Verifies a membership or a non-membership proof, ie it calculates the tree root
+ * based on an entry or matching entry and all siblings and compares that calculated root
+ * with the root that is passed to this function.
+ * @param entry Contains key and value of an entry: [key, value]
+ * @param matching_entry Contains [key, value] of a matching entry only for non-membership proofs
+ * @param siblings Contains array of siblings of entry / matching_entry
+ * @param root The expected root of the tree
+ */
+pub fn verify(entry: [Field; 2], matching_entry: [Option<Field>; 2], siblings: [Field; TREE_DEPTH], root: Field) {
+    let mut calculcated_root: Field = 0;
```

Comment:
There are some occurrences of `calculcated_root`.

Is it `calculated_root` instead of `calculcated_root`?

Reply:
Got it. Yes, feel free to fix it if you want. Thanks.

Reply:
Done, thanks!